lib/destroy.js

```diff
@@ -28,6 +42,6 @@ function destroy (state) {
 
     delete state.account
 
-    return clone(accountProperties)
-  })
+    return Promise.resolve(clone(accountProperties))
+  }
 }
```

Comment:
You can simplify this code by creating a promise variable and then it with the request only if the user is signed in. Like so

```js
var promise = Promise.resolve()

if (internals.isSignedIn(state)) {
  promise = promise.then(function () {
    internals.request({
      method: 'DELETE',
      url: state.url + '/session/account',
      headers: {
        authorization: 'Session ' + state.account.session.id
      }
    })
  })
}

return promise.then(function () {
  internals.clearSession({
    cacheKey: state.cacheKey
  })

  state.emitter.emit('signout', clone(state.account))
  state.emitter.emit('destroy', clone(state.account))

  delete state.account

  return clone(accountProperties)
})
```

Does that make sense?